### ISSUE TITLE
"cannot find -lasound" (compiling on Lubuntu for Windows)

### ISSUE THREAD

**Issue by T0NIT0RMX** [LMMS/lmms#2951]
Hi,

I'm trying to compile LMMS on Lubuntu 14.04 for Windows,
I followed -> https://github.com/LMMS/lmms/wiki/Compiling-lmms-(Windows)

But when I get to step "Package"
I type :
`make package`

And when it's complete, I get :
![lasound](https://cloud.githubusercontent.com/assets/12613367/17272364/e5bd8efa-5694-11e6-9a02-0c5555fbcdff.PNG)

I installed all dependencies, I'm not the only one to have this issue, it also happened to @Umcaruje 

I hope I provided all needed infos, if you need more, just ask and I'll reply ASAP.
Thanks by advance for your help.

**Comment by zonkmachine** [comment]
try:
`sudo apt-get install libasound2-dev`

**Comment by Umcaruje** [comment]
Actually, this is building with mingw, which doesn't provide alsa, and alsa is disabled in the cmake configuration. I'm not sure what is wrong, but I'm expiriencing this same issue.

**Comment by tresf** [comment]
AFAIK, `lmms.exe` is build from the `src/CMakeLists.txt` which explicitly links against ALSA if it's found and [stores it in a variable called `LMMS_REQUIRED_LIBS`](https://github.com/zonkmachine/lmms/blob/master/src/CMakeLists.txt#L121).

I agree with @Umcaruje, this should be empty but AFAIK the issue should happen if running `make` or `make package`.

If you take `${ASOUND_LIBRARY}` out of `LMMS_REQUIRED_LIBS` does it succeed?  If so, there may be something fishing going on with [`FindAlsa.cmake`](https://github.com/LMMS/lmms/blob/master/cmake/modules/FindAlsa.cmake) (or something residual from a previous build, perhaps).

**Comment by T0NIT0RMX** [comment]
@zonkmachine Thanks, but already installed.

@tresf : I think I got it to work thanks to your help
I proceeded like this :
In `src/CMakeLists.txt` I took `${ASOUND_LIBRARY}` out of `LMMS_REQUIRED_LIBS`

**Comment by Umcaruje** [comment]
Ok, so the actual issue seems to be with portaudio, as `asound` seems to be its required library(This is from the CMakeCache.txt):
![screenshot from 2016-08-18 22 25 55](https://cloud.githubusercontent.com/assets/6282045/17791177/357f0cde-659a-11e6-91e2-cc39e78b110c.png)

Anyways, after editing the mingw script to build with `-DWANT_PORTAUDIO=OFF`, lmms built successfully and generated a package with no problems.

**Comment by jasp00** [comment]
The problem may be in the call to `pkg-config`, because the `mingw32-x-portaudio` required libraries are, according to `portaudio-2.0.pc`:
`Libs: -L${libdir} -lportaudio  -lwinmm -lm -ldsound -lole32 -lwinmm -lm  -lsetupapi -lole32 -lwinmm -lm -lole32  -lwinmm -lm -lole32`

**Comment by Umcaruje** [comment]
Closed by #3369

**Comment by tresf** [comment]
> Closed by #3369

Hmm... I don't think so. I'm running into this again with Ubuntu 12.04.  Even with #3369 merged in the error still occurs.

After reading @jasp00's comments, I did a comparison of the Ubuntu 12.04 pkg-config files in Linux versus Mingw.

Notice the Linux version mentions `-lasound` where's Mingw version does not.  Any other ideas?

```diff
diff /opt/mingw32/lib/pkgconfig/portaudio-2.0.pc /usr/lib/x86_64-linux-gnu/pkgconfig/portaudio-2.0.pc
1c1
< prefix=/opt/mingw32
---
> prefix=/usr
3c3
< libdir=${exec_prefix}/lib
---
> libdir=${prefix}/lib/x86_64-linux-gnu
11,12c11,12
< Libs: -L${libdir} -lportaudio  -lwinmm -lm -ldsound -lole32 -lwinmm -lm  -lsetupapi -lole32 -lwinmm -lm -lole32  -lwinmm -lm -lole32 
< Cflags: -I${includedir} -mthreads
---
> Libs: -L${libdir} -lportaudio  -lasound -lm -lpthread
> Cflags: -I${includedir} -pthread
```

I can also confirm that `-DWANT_PORTAUDIO=OFF` corrects the problem, or alternately:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5750da4..ef96704 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,12 +82,14 @@ ENDIF(LMMS_BUILD_APPLE)
 IF(LMMS_BUILD_WIN32)
        SET(WANT_ALSA OFF)
        SET(WANT_JACK OFF)
+       SET(WANT_PORTAUDIO OFF)
        SET(WANT_PULSEAUDIO OFF)
        SET(WANT_SOUNDIO OFF)
        SET(WANT_WINMM ON)
        SET(LMMS_HAVE_WINMM TRUE)
        SET(STATUS_ALSA "<not supported on this platform>")
        SET(STATUS_JACK "<not supported on this platform>")
+       SET(STATUS_PORTAUDIO "<has unmet dependencies>")
        SET(STATUS_PULSEAUDIO "<not supported on this platform>")
        SET(STATUS_SOUNDIO "<disabled in this release>")
        SET(STATUS_WINMM "OK")
```
